tests/libcxxthrd/enc/enc.cpp

```diff
@@ -80,6 +83,13 @@ static int _pthread_create_hook(
     _key_to_thread_id_map.emplace(enc_key, *enc_thread);
     _release_lock(&_enc_lock);
 
+    if (_next_enc_thread_id > (int)(MAX_ENC_KEYS - 1))
```

Comment:
Since you are doing
enc_key = ++_next_enc_thread_id;
the keys start from 1. Therefore you support only max-1 threads as opposed to max threads.